test/bin/generate_common_versions.py

```diff
@@ -0,0 +1,253 @@
+
+    ``get_candidate_repo`` returns the URL of the EC/RC repository for the specified minor version.
+    """
+    if dev_preview:
+        return f"https://mirror.openshift.com/pub/openshift-v4/{ARCH}/microshift/ocp-dev-preview/latest-4.{minor}/el9/os"
+    else:
+        return f"https://mirror.openshift.com/pub/openshift-v4/{ARCH}/microshift/ocp/latest-4.{minor}/el9/os"
+
+
+def get_beta_repo(minor):
+    """
+    :param minor: the minor version, e.g. 19. for 4.19
+
+    ``get_beta_repo`` returns the URL of the beta repository for the specified minor version.
+    If the repo for the wanted version does not exist or it does not provide the necessary
+    packages, it looks for previous releases, for up to three previous minor versions.
+    """
+    for i in range(minor, minor-3, -1):
+        url = f"https://mirror.openshift.com/pub/openshift-v4/{ARCH}/dependencies/rpms/4.{i}-el9-beta"
+        if repo_exists(url) and provides_pkg(url, "cri-o"):
+            return f'"{url}"'
+
+    return None
+
+
+def provides_pkg(repo, pkg):
+    """
+    :param repo: the repository to check
+    :param pkg: the package to look for
+
+    ``provides_pkg`` checks if the repository provides the package specified by `pkg`.
+    """
+    if repo.startswith("https"):
+        try:
+            temp = f"this,{repo}"
+            _ = subprocess.run(
+                    ['dnf', 'repoquery', pkg, '--quiet',
+                     '--disablerepo', '*', '--repofrompath', temp],
+                    capture_output=True,
+                    check=True
+                )
+
+            return True
+
+        except subprocess.CalledProcessError:
+            return False
+    else:
+        try:
+            _ = subprocess.run(
+                    ['dnf', 'repoquery', pkg, '--quiet',
+                     '--repo', repo],
+                    capture_output=True,
+                    check=True
+            )
+
+            return True
+
+        except subprocess.CalledProcessError:
+            return False
+
+
+def dnf_repo_is_enabled(repo_name):
+    """
+    :param repo_name: the name of the repository to check
+
+    ``dnf_repo_is_enabled`` checks if the repository is enabled on this system.
+    """
+    try:
+        result = subprocess.run(
+            ["dnf", "repolist"],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.DEVNULL,
+            text=True,
+            check=True
+        )
+        return repo_name in result.stdout
+    except subprocess.CalledProcessError:
+        return False
+
+
+def repo_exists(repo_url):
+    """
+    :param repo_url: the URL of a repository
+
+    ``repo_exists`` checks if a URL points to a valid repository.
+    """
+    url = repo_url + "/repodata/repomd.xml"
+    r = requests.get(url)
+    if r.status_code == 404:
+        return False
+    else:
+        return True
+
+
+def get_sub_repo(minor):
```

Comment:
I think we can use full name and hint on the functionality, e.g. `get_subscription_repo_name_if_exists`